packages/repository/src/model.ts

```diff
@@ -249,18 +257,36 @@ export abstract class Model {
 
   /**
    * Convert to a plain object as DTO
+   *
+   * If `ignoreUnknownProperty` is set to false, convert all properties in the
+   * model instance, otherwise only convert the ones defined in the model
+   * definitions.
+   *
+   * See function `asObject` for each property's conversion rules.
    */
   toObject(options?: Options): Object {
-    let obj: AnyObject;
+    const obj: AnyObject = {};
+
     if (options && options.ignoreUnknownProperties === false) {
-      obj = {};
       for (const p in this) {
         const val = (this as AnyObject)[p];
         obj[p] = asObject(val, options);
       }
-    } else {
-      obj = this.toJSON();
+      return obj;
     }
+
+    const def = (<typeof Model>this.constructor).definition;
```

Comment:
_A nitpick, feel free to ignore._

I believe we prefer `as` casts over `<>`?

```suggestion
    const def = (this.constructor as typeof Model).definition;
```

Reply:
yep agreed, and I also refactored the one in `toJson`  from `<>` to `as`